src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -4541,6 +4541,11 @@ public boolean isStarting()
         return operationMode == Mode.STARTING;
     }
 
+    public boolean isJoining()
+    {
+        return operationMode == Mode.JOINING;
+    }
```

Comment:
we should also guard during waiting_to_finish_bootstrap

Reply:
I'm debating with myself if we want to expose a separate function for this but given that we only care about joining and waiting_to_finish_bootstrap together within a single context outside of SS, I'll refactor the existing isJoining to isJoiningOrWaitingToFinishBootstrap